packages/nimbus/src/components/money-input/money-input.tsx

```diff
@@ -283,38 +269,30 @@ export const MoneyInputComponent = (props: MoneyInputProps) => {
     [handleCurrencyChange]
   );
 
-  const stateProps = {
-    isInvalid,
-    isDisabled,
-    isReadOnly,
-    hasFocus,
-  };
+  const noCurrenciesLabelId = useId();
+  const highPrecisionBadgeId = useId();
 
   return (
-    <MoneyInputRootSlot
-      {...recipeProps}
-      {...styleProps}
-      {...stateProps}
-      {...remainingProps}
-    >
+    <MoneyInputRootSlot {...recipeProps} {...styleProps} {...remainingProps}>
       <MoneyInputContainerSlot>
         {/* Currency Select or Label */}
-        <MoneyInputCurrencySelectSlot {...stateProps}>
+        <MoneyInputCurrencySelectSlot>
           {hasNoCurrencies ? (
             <MoneyInputCurrencyLabelSlot asChild>
-              <label data-testid="currency-label">{value.currencyCode}</label>
+              <label id={noCurrenciesLabelId}>{value.currencyCode}</label>
```

Comment:
giving the label and `id` and associating it to the amount input with `aria-labelledby` is better than using `htmlFor` - `htmlFor` will completely replace any other `aria-label`, etc with the text of the label `htmlFor` is set on.  

So if the amount input's `aria-label` is `amount` and the label text is `usd`, with `aria-labelledby` the input's label will be `usd amount`, with `htmlFor` the inputs label will be `usd`.